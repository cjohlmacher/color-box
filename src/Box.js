import './Box.css';

const Box = ({backgroundColor, width, height, deleteBox}) => {
    const boxStyles = {
        backgroundColor,
        width: width+'px',
        height: height+'px'
    };
    
    const handleDelete = () => {
        deleteBox();
    };

    return (
        <>
            <div className="Box" data-testid="box" style={boxStyles}></div>
            <button name="delete" onClick={handleDelete}>X</button>
        </>
    )
};

export default Box;