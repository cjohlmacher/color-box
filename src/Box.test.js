import { render } from '@testing-library/react';
import React from 'react';
import Box from './Box';

it('should render a  component', () => {
    render(<Box backgroundColor="aqua" width={50} height={40} deleteBox={() => console.log('Deleted')}/>);
});

it('should match snapshot', () => {
    const {asFragment} = render(<Box backgroundColor="aqua" width={50} height={40} deleteBox={() => console.log('Deleted')}/>);
    expect(asFragment()).toMatchSnapshot();
});