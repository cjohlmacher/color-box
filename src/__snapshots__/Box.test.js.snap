// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="Box"
    data-testid="box"
    style="background-color: aqua; width: 50px; height: 40px;"
  />
  <button
    name="delete"
  >
    X
  </button>
</DocumentFragment>
`;
